### ISSUE TITLE
Rustdoc should report unknown `doc(...)` attributes

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#82652]
I tried this code:

```rust
#![doc(xxx)]
```

I expected to see this happen: Rustdoc warns that `xxx` is an unknown doc attribute.
Instead, this happened: Rustdoc silently ignores the attribute.

### Meta

`rustdoc --version`: rustdoc 1.52.0-nightly (0148b971c 2021-02-18)

Found while working on https://github.com/rust-lang/rust/pull/82651.

**Comment by GuillaumeGomez** [comment]
Oh funny! Gonna fix it then. :)
